arcgis-ios-sdk-samples/Utility network/Create load report/CreateLoadReportViewController.swift

```diff
@@ -0,0 +1,360 @@
+        case .included:
+            cell = tableView.dequeueReusableCell(withIdentifier: Section.included.cellIdentifier, for: indexPath)
+            phase = includedPhases[indexPath.row]
+            cell.textLabel?.text = String(format: "Phase: %@", phase.name)
+            if let summary = summaries[phase] {
+                let formattedString = String(format: "C: %@  L: %@", numberFormatter.string(from: NSNumber(value: summary.totalCustomers))!, numberFormatter.string(from: NSNumber(value: summary.totalLoad))!)
+                cell.detailTextLabel?.text = formattedString
+            } else {
+                cell.detailTextLabel?.text = "N/A"
+            }
+        case .excluded:
+            cell = tableView.dequeueReusableCell(withIdentifier: Section.excluded.cellIdentifier, for: indexPath)
+            phase = excludedPhases[indexPath.row]
+            cell.textLabel?.text = phase.name
+        }
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCell.EditingStyle {
+        switch Section.allCases[indexPath.section] {
+        case .included:
+            return .delete
+        case .excluded:
+            return .insert
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        /// Binary search to find the insertion index in a sorted array.
+        func index(forInserting phase: AGSCodedValue, into array: [AGSCodedValue]) -> Int {
+            (array as NSArray).index(
+                of: phase,
+                inSortedRange: NSRange(array.indices),
+                options: [.insertionIndex],
+                usingComparator: { ($0 as! AGSCodedValue).name.compare(($1 as! AGSCodedValue).name) }
+            )
+        }
+        
+        tableView.beginUpdates()
+        defer { tableView.endUpdates() }
+        tableView.deleteRows(at: [indexPath], with: .automatic)
+        
+        let insertionIndex: Int
+        switch editingStyle {
+        case .delete:
+            let phase = includedPhases[indexPath.row]
+            includedPhases.remove(at: indexPath.row)
+            insertionIndex = index(forInserting: phase, into: excludedPhases)
+            excludedPhases.insert(phase, at: insertionIndex)
+            // Remove the summary.
+            summaries.removeValue(forKey: phase)
+            tableView.insertRows(at: [IndexPath(row: insertionIndex, section: .excluded)], with: .automatic)
+        case .insert:
+            let phase = excludedPhases[indexPath.row]
+            excludedPhases.remove(at: indexPath.row)
+            insertionIndex = index(forInserting: phase, into: includedPhases)
+            includedPhases.insert(phase, at: insertionIndex)
+            tableView.insertRows(at: [IndexPath(row: insertionIndex, section: .included)], with: .automatic)
+        default:
+            return
```

Comment:
I don't remember when (if ever) this method is called with `none`, but that might be a possibility. If it does happen, the row at the given index path will be deleted--not great. It would be a good idea to guard either that `editingStyle != .none`, or, even better (since the enum is not frozen) `editingStyle == .delete && editingStyle == .insert`. Another option would be to use `performBatchUpdates(_:completion:)` inside each of the cases. And come to think of it, that is exactly what the documentation for `beginUpdates()` recommends:

![image](https://user-images.githubusercontent.com/2257493/112864856-16020c00-906d-11eb-853c-985c9ce2e302.png)

I'm sorry for not remembering that sooner.